src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ManagedThreadId.cs

```diff
@@ -174,7 +176,7 @@ public ImmutableIdDispenser AllocateId(out int id)
             }
         }
 
-        public const int IdNone = 0;
+        public const int IdNone = -1;
```

Comment:
Was there any actual problem with 0?

Zero is a tiny bit more efficient - the instructions to check for zero tend to have smaller encodings.

Reply:
-1 is so that I could just fetch the ID unchecked on the critical paths and it would automatically go on a slow path if not yet initialized, since it would not fit into 16 bits.

Reply:
Where is this trick used? I am not able to find it.

Reply:
CurrentManagedThreadIdUnchecked is not checking for initialization. 

Not supposed to, unless i mixed up something.

Reply:
Ok, I see that now. Using -1 to signal "not yet initialized" state will make it harder to switch back to thread statics once we get them optimized.

Reply:
Depending how thread statics end up working, I think we may just initialize them to the right value, if that does not entail checking "is initialized" all the time. 
Or try initializing ID early enough, before it can possibly be used in any lock. It may already be the case, but I am not 100% sure and did not want to force such invariant just yet.

Alternatively we may do `-1` when accessing the value.

Reply:
Thread statics are initialized to 0.

The current initialization to -1 in the unmanaged runtime is piggy backing on "is initialized" check done by reverse pinvoke that is the only way to enter managed code in native AOT. Once this is switched back to managed thread static, the initialization to -1 will need the "is initialized" check all the time.

> Alternatively we may do -1 when accessing the value.

Yes, I think it would be better.